src/main/java/uk/gov/justice/laa/crime/cfecrime/utils/FullMeansTestOutcomeCalculator.java

```diff
@@ -14,40 +14,52 @@ private FullMeansTestOutcomeCalculator(){
     }
 
     public static MeansTestOutcome getFullMeansTestOutcome(FullAssessmentResult result, CaseType caseType, MagCourtOutcome magCourtOutcome){
-        log.debug("Get the outcome of the Full Means Test. Inputs: result = {} caseType = {} magCourtOutcome = {}", result, caseType, magCourtOutcome);
-
-        //Fail result: Heard In Magistrates Court
-        Set<CaseType> caseTypesHeardInMagistratesCourt = Set.<CaseType>of(CaseType.COMMITAL, CaseType.SUMMARY_ONLY, CaseType.EITHER_WAY);
+        log.debug("FullMeansTestOutcome start. Inputs: result = {} caseType = {} magCourtOutcome = {}", result, caseType, magCourtOutcome);
 
         MeansTestOutcome meansTestOutcome = null;
-        if (result != null && caseType != null && magCourtOutcome != null) {
 
-            if (result == FullAssessmentResult.PASS) {
-                    //All Eligible with no contribution
+        // Magistrates' court
+        if (caseType == CaseType.SUMMARY_ONLY ||
+            caseType == CaseType.COMMITAL ||
+            (caseType == CaseType.EITHER_WAY && magCourtOutcome != MagCourtOutcome.COMMITTED_FOR_TRIAL && magCourtOutcome != null)) {
+                if (result == FullAssessmentResult.FAIL) {
+                    meansTestOutcome = MeansTestOutcome.INELIGIBLE;
+                }
+                if (result == FullAssessmentResult.PASS) {
                     meansTestOutcome = MeansTestOutcome.ELIGIBLE_WITH_NO_CONTRIBUTION;
-            }
-            if (result == FullAssessmentResult.FAIL) {
-                // Either way" - offence type that could be heard in Magistrates Court or
-                // Crown Court AND magistrate outcome is NOT COMMITTED_FOR_TRIAL
-                if ((caseType == CaseType.EITHER_WAY && magCourtOutcome != MagCourtOutcome.COMMITTED_FOR_TRIAL) ||
-                        caseTypesHeardInMagistratesCourt.contains(caseType)) {
+                }
+        }
+
+        // Crown Court
+        if (caseType == CaseType.INDICTABLE ||
+            caseType == CaseType.CC_ALREADY ||
+            (caseType == CaseType.EITHER_WAY && magCourtOutcome == MagCourtOutcome.COMMITTED_FOR_TRIAL)) {
+                if (result == FullAssessmentResult.INEL) {
                     meansTestOutcome = MeansTestOutcome.INELIGIBLE;
-                }else{
+                }
+                if (result == FullAssessmentResult.FAIL) {
                     meansTestOutcome = MeansTestOutcome.ELIGIBLE_WITH_CONTRIBUTION;
                 }
```

Comment:
Wondering if we should break this logic out into a separate function? I think it might be possible to obtain an INEL from other paths (if no now, then future might not preclude it) so maybe to be defensive we should always convert INEL to INELIGIBLE - then multiple branches could use it. Looks better to classify case first though :-)

Reply:
Always converting INEL to Ineligible sounds like an *assumption*. I'd rather we only process inputs we know the outputs for, and raise an exception for all other circumstances.